README.md

```diff
@@ -55,37 +55,64 @@ Alternatively you can wrap any code which may throw errors using the client's `w
 
 ## Advanced Usage
 
-### Default Annotations
+### Notice Annotations
 
-It's possible to annotate error notices with all sorts of useful information. Below, the various top-level interface methods are listed, along with their effects.
-
-* `airbrake.setEnvironmentName(string)` sets the environment name, e.g. production or staging.
-* `airbrake.addEnvironment(object)` adds environment information reported alongside all errors.
-* `airbrake.addParams(object)` adds params information reported alongside all errors.
-* `airbrake.addSession(object)` adds session information reported alongside all errors.
-* `airbrake.addContext(object)` adds context information reported alongside all errors. This hash is reserved for notifier and Airbrake backend will ignore unknown keys.
-
-Additionally, much of this information can be added to captured errors at the time they're captured by supplying it in the object being reported.
+It's possible to annotate error notices with all sorts of useful information at the time they're captured by supplying it in the object being reported.
 
     try {
       startApp();
     } catch (err) {
-      airbrake.push({
+      airbrake.notify({
```

Comment:
Yeah, one reason is that we must preserve backwards compatibility where possible.

But I also like existing behaviour when you can:
- pass only exception, .e.g. `notify(exception)`
- or notice-like hash, e.g. `notify({error: exception, params: {...}})`

I am not sure if this is possible in Ruby, but in Python I can define notify using named args like this:

```
def notify(exc, params={}, session={}, environment={})
```

and pass only arguments I need, e.g.
- `notify(exc)`
- `notify(exc, params={hello: world})`
- `notify(exc, session={hello: world})`
- `notify(exc, session={hello: world}, environment={env1: value})`

Can you do the same in Ruby?